diff --git a/oaf/src/af_client.c b/oaf/src/af_client.c
index 8f78d48..13e0a3d 100755
--- a/oaf/src/af_client.c
+++ b/oaf/src/af_client.c
@@ -574,7 +574,7 @@ static struct nf_hook_ops af_client_ops[] = {
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 15, 0)
 static void client_timer_handler(struct timer_list *t)
 {
-    af_client_info_t *client = from_timer(client, t, client_timer);
+    af_client_info_t *client = container_of(t, af_client_info_t, client_timer);
 #else
 static void client_timer_handler(unsigned long data)
 {
@@ -623,7 +623,7 @@ static void client_timer_handler(unsigned long data)
         return;
     }
     
-    del_timer_sync(&client->client_timer);
+    timer_delete_sync(&client->client_timer);
 }
 
 
diff --git a/oaf/src/app_filter.c b/oaf/src/app_filter.c
index edc2719..d90be2e 100755
--- a/oaf/src/app_filter.c
+++ b/oaf/src/app_filter.c
@@ -1565,7 +1565,7 @@ void init_oaf_timer(void)
 
 void fini_oaf_timer(void)
 {
-	del_timer_sync(&oaf_timer);
+	timer_delete_sync(&oaf_timer);
 	AF_INFO("del oaf timer...ok");
 }
 
